documentation/tutorials/text-search-semantic.md

```diff
@@ -0,0 +1,317 @@
+
+```
+rank-profile bert_title_body_all inherits default {
+    function dot_product_title() {
+        expression: sum(query(tensor_bert)*attribute(title_bert))
+    }
+    function dot_product_body() {
+        expression: sum(query(tensor_bert)*attribute(body_bert))
+    }
+    first-phase {
+        expression: dot_product_title() + dot_product_body()
+    }
+}
+```
+
+The [rank-profile](../reference/search-definitions-reference.html#rank-profile) `bert_title_body_all` will sort all the matched documents according to the sum of the dot-products between query and title and query and body vectors. Different rank-profiles can be defined for experimentation.
+
+### ANN operator
+
+Once that query and document tensors as well as rank-profiles that uses them are all defined, it is possible to use the embeddings to match and to rank the documents by using the `nearestNeighbor` operator together with the appropriate rank-profile.
+
+```
+{
+	"yql":"select * from sources * where ([{"targetNumHits": 1000, "label": "nns"}]nearestNeighbor(title_bert, tensor_bert));"
+	"userQuery":"what types of plate boundaries cause deep sea trenches"
+	"ranking":{
+		"profile":"bert_title_body_all"
+		"listFeatures":"true"
+	}
+	"ranking.features.query(tensor_bert)":"[0.05121087115032622, -0.0035218095295999675, ..., 0.05303904445092506]"
+	...
+} 
+```
+
+The query above uses the `nearestNeighbor` operator to match documents based on the euclidean distance between the title embedding (`title_bert`) and the query embedding (`tensor_bert`). It is possible to annotate the `nearestNeighbor` with properties such as `targetNumHits` that defines the target number of documents to be matched. In addition, we specificy that the matched documents will be ranked by the `bert_title_body_all` rank-profile. 
+
+
+## ANN results
+
+The table below show results obtained by matching the closest 1.000 document vectors to the query vector in terms of the Euclidean distance. Even though Vespa support approximate nearest neighbor search, we set the method to be brute force to remove the approximation error from the analysis in this tutorial. This means that the documents matched were indeed the closest ones to the query. The `ANN(title, bert)` in the table below means that we matched documents by comparing the document title embedding to the query embedding where the embeddings were created by the sentence BERT model. 
+
+All the results involving embeddings in this tutorial are generated via the sentence BERT model. The results obtained with the Universal Sentence Encoder model were very similar and therefore omitted. On the other hand, the results obtained with the Word2Vec model were way worse than expected and were left out because they belong to a more post-mortem type of discussion.
```

Comment:
Last sentence here, probably just drop the post-mortem and mention that Word2Vec underperformed significantly and thus omitted. Ideally, a sentence on why you think that is?